File: src/main/java/ordilov/randomplay/playlist/domain/youtube/YoutubePlaylistItems.java
package ordilov.randomplay.playlist.domain.youtube;

import java.util.List;
import lombok.Getter;

@Getter
public class YoutubePlaylistItems {
  private String kind;
  private String etag;
  private PageInfo pageInfo;
  private String nextPageToken;
  private String prevPageToken;
  private List<YoutubePlaylistItem> items;

  @Getter
  private static class ContentDetails{
    private String note;
    private String endAt;
    private String startAt;
    private String videoId;
  }

  @Getter
  private static class Status{
    private String privacyStatus;
  }
}

File: src/main/java/ordilov/lissn/playlist/domain/youtube/YoutubePlaylistItems.kt
package ordilov.lissn.playlist.domain.youtube

data class YoutubePlaylistItems(
    val kind: String,
    val etag: String,
    val pageInfo: PageInfo,
    val nextPageToken: String,
    val prevPageToken: String,
    val items: List<YoutubePlaylistItem>,
)

data class ContentDetails(
    val note: String,
    val endAt: String,
    val startAt: String,
    val videoId: String
)

data class Status(
    val privacyStatus: String
)